src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx

```diff
@@ -18,6 +20,11 @@ import { parseUnits } from "ethers/lib/utils";
 import { decodeAddress } from "@polkadot/util-crypto";
 
 import { hasTokenSupplies, getPriceCompatibility } from "./helpers";
+import {
```

Comment:
Let's wrap this code into 1 more abstraction to get rid of substrate inside the EVMAdaptor.

Reply:
Why?

Reply:
Decided to move this logic to NetworkManager.

Reply:
Resolved

Reply:
Which exact line you mean?

Reply:
It was a snapshot from the previous commit. All is good now.